src/htm/regions/InputRegion.hpp

```diff
@@ -0,0 +1,80 @@
+/* ---------------------------------------------------------------------
+ * HTM Community Edition of NuPIC
+ * Copyright (C) 2019, Numenta, Inc.
+ *
+ * Author: David Keeney, Nov. 2019   dkeeney@gmail.com
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses.
+ * --------------------------------------------------------------------- */
+
+/** @file
+ * Implementation of a network region that handles direct input to a region.
```

Comment:
nit: for a while I was wondering if each Encoder*Region is an InputRegion (which is not, per se). So maybe DimensionsRegion?

Reply:
I am not in love with the name InputRegion either.
Although it is a minimal region it is sort of part of the infrastructure rather than one of the regions that can be used as a building block.  DimensionsRegion does not really describe it either.  

Its a facade to hang a buffer that is used for manually setting a target region's input buffer.
I could call it InputFacade  and move it to the engine folder.

Reply:
Maybe "DirectInput".  The feature is for directly setting the input but the way it is used its more of an indirect input.  Any other ideas?

Reply:
RawInput ?

Reply:
> DimensionsRegion does not really describe it either.
Its a facade to hang a buffer that is used for manually setting a target region's input buffer.

Ok, it internally sets the input buffers. But from user's perspective, isn't the only usecase defining the dimensions? 

>  and move it to the engine folder. 

:+1: for the move to engine, so users don't see it among the "normal" Regions for use. Maybe even make its constructor private or something - as to avoid that anyone actually tries to create a region InputRegion. As I understand your HOWTOs, we'll only use it through the link with referring the "INPUT" name.

Reply:
Ok, "RawInput" it is (thank you @Lupino ) and I will move it to the engine.